manual/spam_challenge.php

```diff
@@ -27,44 +27,40 @@ function print_prefix($name, $a, $b) {
 	return "$name($a, $b)";
 }
 
-$challenges = array(
+const CHALLENGES = array(
 	// name, print, generator
 	array('max',   'print_prefix'),
 	array('min',   'print_prefix'),
 	array('minus', 'print_infix', 'gen_minus'),
 	array('plus',  'print_infix', 'gen_plus'),
 );
 
-
 // generate a challenge
 function gen_challenge() {
-	global $challenges, $nums;
-	$c = $challenges[rand(0, count($challenges)-1)];
+    $c = CHALLENGES[rand(0, count(CHALLENGES)-1)];
```

Comment:
I think we should stick with the tab indentation *for now*:
```suggestion
	$c = CHALLENGES[rand(0, count(CHALLENGES)-1)];
```